kloppy/infra/serializers/event/wyscout/deserializer_v3.py

```diff
@@ -547,12 +616,40 @@ def deserialize(self, inputs: WyscoutInputs) -> EventDataset:
                         result=None,
                         qualifiers=_generic_qualifiers(raw_event),
                         event_name=raw_event["type"]["primary"],
-                        **generic_event_args
+                        **generic_event_args,
                     )
 
                 if event and self.should_include_event(event):
                     events.append(transformer.transform_event(event))
 
+                if next_event:
+                    event_formation_change_info = (
+                        identify_artificial_formation_change_event(
+                            raw_event, next_event, teams, home_team, away_team
+                        )
+                    )
+                    for (
+                        formation_change_team,
+                        formation_change_event_kwargs,
+                    ) in event_formation_change_info.items():
+                        generic_event_args.update(
+                            {
+                                "event_id": None,
```

Comment:
It would be good to give this event an id. I checked the Metrica implementation which also creates synthetic events and it seems it's reusing the original event id. 

The eventIds should be unique over a match. Maybe you can add a prefix to the original eventId and use that one? 

And would be good to use the term 'synthetic' instead of 'artificial'.